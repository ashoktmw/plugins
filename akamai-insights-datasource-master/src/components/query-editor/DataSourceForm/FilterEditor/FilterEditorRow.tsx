import { SelectableValue } from '@grafana/data';
import { InlineField, InlineFieldRow, Select } from '@grafana/ui';
import React, { JSX } from 'react';

import '../../../../styles/styles.css';
import { FilterEditor } from './FilterEditor';
import { FormService } from '../../../../services/form.service';
import { DiscoveryApiModel } from '../../../../types/discovery-api.model';
import {
  FilterFormModel,
  FilterType,
  filterTypeOptions,
  metricOperatorOptions,
  shortLabelWidth
} from '../FormTypes';

interface FilterEditorProps {
  model: FilterFormModel;
  discoveryApiModel: DiscoveryApiModel;
  dimensions: SelectableValue<string>[];
  metrics: SelectableValue<string>[];
  onChange: (value: FilterFormModel) => void;
  style?: Partial<CSSStyleDeclaration>
}

function getFilterEditor(type: string | undefined,
                         dimensions: SelectableValue<string>[],
                         metrics: SelectableValue<string>[],
                         model: FilterFormModel,
                         discoveryApiModel: DiscoveryApiModel,
                         onChange: (query: FilterFormModel) => void): JSX.Element {
  switch (type) {
    case FilterType.Dimension:
      return (
        <FilterEditor
          model={model.query}
          names={dimensions}
          operators={FormService.getDimensionsOperatorsOptions(model, discoveryApiModel)}
          multiExpressions={true}
          onChange={query => onChange({ ...model, query })}
        />
      );
    case FilterType.Metric:
      return (
        <FilterEditor
          model={model.query}
          names={metrics}
          operators={metricOperatorOptions}
          multiExpressions={false}
          onChange={query => onChange({ ...model, query })}
        />
      );
    default:
      return <InlineField><span>Not implemented yet</span></InlineField>;
  }
}

export function FilterEditorRow({ model, discoveryApiModel, dimensions, metrics, onChange }: FilterEditorProps): JSX.Element {
  const Editor = getFilterEditor(model.type?.value, dimensions, metrics, model, discoveryApiModel, onChange);

  const changeFilterTypeQuery = (type: SelectableValue<string>): void => {
    let updatedModel = {} as FilterFormModel;

    if (type.value === FilterType.Dimension) {
      updatedModel = FormService.creteEmptyFilter(dimensions[ 0 ]);
    } else if (type.value === FilterType.Metric) {
      updatedModel = FormService.creteEmptyFilter(metrics[ 0 ]);
    }

    onChange({ ...updatedModel, type });
  };

  return (
    <InlineFieldRow className="padding-left-20">
      <InlineField>
        <Select
          isClearable={false}
          width={shortLabelWidth}
          options={filterTypeOptions}
          value={model.type}
          onChange={changeFilterTypeQuery}
        />
      </InlineField>

      {Editor}
    </InlineFieldRow>
  );
}
