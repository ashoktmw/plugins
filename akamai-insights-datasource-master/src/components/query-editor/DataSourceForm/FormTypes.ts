import { SelectableValue } from '@grafana/data';
import { groupBy } from 'lodash';

import { enumToSelectableValues, prettyEnum, stringsToSelectableValues } from '../../../utils/utils';

export const shortLabelWidth = 25;
export const mediumLabelWidth = 50;

export enum FilterType {
  Dimension = 'Dimension',
  Metric = 'Metric'
}

export const filterTypeOptions = enumToSelectableValues(FilterType);

export const filterTypeOptionsMap = groupBy(filterTypeOptions, 'value');

export interface FilterFormModel {
  type: SelectableValue<string>;
  query: FilterQueryFormModel;
}

export interface FilterQueryFormModel {
  name?: SelectableValue<string>;
  operator?: SelectableValue<string>;
  expressions?: string[];
}

export enum DimensionOperator {
  InList = 'IN_LIST',
  MatchRegExp = 'MATCH_REGEXP',
  NotMatchRegExp = 'NOT_MATCH_REGEXP',
  BeginsWith = 'BEGINS_WITH',
  NotBeginsWith = 'NOT_BEGINS_WITH',
  EndsWith = 'ENDS_WITH',
  NotEndsWith = 'NOT_ENDS_WITH',
  Contains = 'CONTAINS',
  NotContains = 'NOT_CONTAINS',
  Exact = 'EXACT'
}

export enum FilterValueType {
  Text = 'TEXT',
  Enum = 'ENUM'
}

export const dimensionOperatorOptions = enumToSelectableValues(DimensionOperator);

export const authorizableOrEnumFiltersOperatorOptions = stringsToSelectableValues([ DimensionOperator.InList ], prettyEnum);

export enum MetricOperator {
  Equal = 'EQUAL',
  NotEqual = 'NOT_EQUAL',
  LessThan = 'LESS_THAN',
  GreaterThan = 'GREATER_THAN'
}

export const metricOperatorOptions = enumToSelectableValues(MetricOperator);

export interface SortByFormModel {
  type?: SelectableValue<string>;
  query: SortByQueryFormModel;
}

export interface SortByQueryFormModel {
  name?: SelectableValue<string>;
  sortOrder?: SelectableValue<string>;
}

export enum SortByOrder {
  Ascending = 'ASCENDING',
  Descending = 'DESCENDING'
}

export const sortByOrderOptions = enumToSelectableValues(SortByOrder);
