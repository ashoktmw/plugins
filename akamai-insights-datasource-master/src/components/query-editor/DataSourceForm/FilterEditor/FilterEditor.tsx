import { SelectableValue } from '@grafana/data';
import { InlineField, Input, Select, TagsInput } from '@grafana/ui';
import { first } from 'lodash';
import React, { Fragment, JSX } from 'react';

import { splitByComma } from '../../../../utils/utils';
import { FilterQueryFormModel, mediumLabelWidth, shortLabelWidth } from '../FormTypes';

export interface FilterQueryEditorProps {
  model: FilterQueryFormModel;
  names: SelectableValue<string>[];
  operators: SelectableValue<string>[];
  multiExpressions: boolean;
  onChange: (query: FilterQueryFormModel) => void;
}

export function FilterEditor({ model, names, operators, multiExpressions, onChange }: FilterQueryEditorProps): JSX.Element {
  const ExpressionField = multiExpressions ?
    <InlineField>
      <TagsInput
        placeholder="Expressions"
        width={mediumLabelWidth}
        tags={model.expressions}
        onChange={expressions => onChange({ ...model, expressions: splitByComma(expressions) })}
      />
    </InlineField> :
    <InlineField>
      <Input
        placeholder="Expression"
        width={mediumLabelWidth}
        value={first(model.expressions)}
        onChange={event => onChange({ ...model, expressions: [ (event.target as any).value ] })}
      />
    </InlineField>;

  return (
    <Fragment>
      <InlineField>
        <Select
          placeholder="Name"
          isClearable={false}
          options={names}
          width={shortLabelWidth}
          value={model.name}
          onChange={name => onChange({ ...model, name })}
        />
      </InlineField>

      <InlineField>
        <Select
          placeholder="Operator"
          isClearable={false}
          width={shortLabelWidth}
          options={operators}
          value={model.operator}
          onChange={operator => onChange({ ...model, operator })}
        />
      </InlineField>

      {ExpressionField}
    </Fragment>
  );
}
