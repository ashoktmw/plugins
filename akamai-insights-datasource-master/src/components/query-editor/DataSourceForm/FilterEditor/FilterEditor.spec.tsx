import React from 'react';
import renderer from 'react-test-renderer';

import { FilterEditor } from './FilterEditor';
import { mockDimensionsOptions, mockMetricsOptions } from '../../../../test/mocks/mock-selectable-values';
import { dimensionOperatorOptions, FilterQueryFormModel, metricOperatorOptions } from '../FormTypes';

describe('FilterEditor component', () => {
  it('should render dimension editor', () => {
    const model: FilterQueryFormModel = {
      name: mockDimensionsOptions[ 1 ],
      operator: dimensionOperatorOptions[ 0 ],
      expressions: [ '1234', '5678' ]
    };

    const names = mockDimensionsOptions;
    const onChange = jest.fn();

    const tree = renderer.create(
      <FilterEditor model={model} names={names} operators={dimensionOperatorOptions} multiExpressions={true} onChange={onChange}/>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render metric editor', () => {
    const model: FilterQueryFormModel = {
      name: mockMetricsOptions[ 1 ],
      operator: metricOperatorOptions[ 0 ],
      expressions: [ '1000' ]
    };

    const names = mockMetricsOptions;
    const onChange = jest.fn();

    const tree = renderer.create(
      <FilterEditor model={model} names={names} operators={metricOperatorOptions} multiExpressions={false} onChange={onChange}/>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
