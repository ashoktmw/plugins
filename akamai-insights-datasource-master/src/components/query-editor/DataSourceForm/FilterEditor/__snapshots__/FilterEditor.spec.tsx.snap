// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterEditor component should render dimension editor 1`] = `
[
  <div
    className="css-1f5gueu"
  >
    <div
      className="css-1rfunm5"
    >
      <div
        className="css-jjhwg3-input-wrapper css-rmjxhr-SelectContainer"
        onKeyDown={[Function]}
      >
        <span
          className="css-1f43avz-a11yText-A11yText"
          id="react-select-2-live-region"
        />
        <span
          aria-atomic="false"
          aria-live="polite"
          aria-relevant="additions text"
          className="css-1f43avz-a11yText-A11yText"
        />
        <div
          className="css-1scgfe8"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
        >
          <div
            className="css-1kl463j-grafana-select-value-container"
          >
            <div
              className="css-18bh10p-singleValue css-upz218-SingleValue"
            >
              time1Hour
            </div>
            <div
              className=" css-lq7btu-Input"
              data-value=""
            >
              <input
                aria-autocomplete="list"
                aria-expanded={false}
                aria-haspopup={true}
                autoCapitalize="none"
                autoComplete="off"
                autoCorrect="off"
                className=""
                disabled={false}
                id="react-select-2-input"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                role="combobox"
                spellCheck="false"
                style={
                  {
                    "background": 0,
                    "border": 0,
                    "color": "inherit",
                    "font": "inherit",
                    "gridArea": "1 / 2",
                    "label": "input",
                    "margin": 0,
                    "minWidth": "2px",
                    "opacity": 1,
                    "outline": 0,
                    "padding": 0,
                    "width": "100%",
                  }
                }
                tabIndex={0}
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            className="css-uvldi-input-suffix"
          >
            <div
              className="css-wf08df-Icon"
            >
              <svg
                data-testid="angle-down"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="css-1f5gueu"
  >
    <div
      className="css-1rfunm5"
    >
      <div
        className="css-jjhwg3-input-wrapper css-rmjxhr-SelectContainer"
        onKeyDown={[Function]}
      >
        <span
          className="css-1f43avz-a11yText-A11yText"
          id="react-select-3-live-region"
        />
        <span
          aria-atomic="false"
          aria-live="polite"
          aria-relevant="additions text"
          className="css-1f43avz-a11yText-A11yText"
        />
        <div
          className="css-1scgfe8"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
        >
          <div
            className="css-1kl463j-grafana-select-value-container"
          >
            <div
              className="css-18bh10p-singleValue css-upz218-SingleValue"
            >
              In list
            </div>
            <div
              className=" css-lq7btu-Input"
              data-value=""
            >
              <input
                aria-autocomplete="list"
                aria-expanded={false}
                aria-haspopup={true}
                autoCapitalize="none"
                autoComplete="off"
                autoCorrect="off"
                className=""
                disabled={false}
                id="react-select-3-input"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                role="combobox"
                spellCheck="false"
                style={
                  {
                    "background": 0,
                    "border": 0,
                    "color": "inherit",
                    "font": "inherit",
                    "gridArea": "1 / 2",
                    "label": "input",
                    "margin": 0,
                    "minWidth": "2px",
                    "opacity": 1,
                    "outline": 0,
                    "padding": 0,
                    "width": "100%",
                  }
                }
                tabIndex={0}
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            className="css-uvldi-input-suffix"
          >
            <div
              className="css-wf08df-Icon"
            >
              <svg
                data-testid="angle-down"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="css-1f5gueu"
  >
    <div
      className="css-1rfunm5"
    >
      <div
        className="css-dvq0w4"
      >
        <div
          className="css-xcstkt-input-wrapper"
          data-testid="input-wrapper"
        >
          <div
            className="css-1w5c5dq-input-inputWrapper"
          >
            <input
              className="css-1n6y0bv-input-input"
              onBlur={[Function]}
              onChange={[Function]}
              onKeyDown={[Function]}
              placeholder="Expressions"
              style={
                {
                  "paddingLeft": undefined,
                  "paddingRight": 12,
                }
              }
              value=""
            />
            <div
              className="css-7y3u6k-input-suffix"
            >
              <button
                className="css-m5e3oq-button"
                disabled={true}
                onClick={[Function]}
                type="button"
              >
                <span
                  className="css-1mhnkuh"
                >
                  Add
                </span>
              </button>
            </div>
          </div>
        </div>
        <ul
          className="css-1pmkqzh"
        >
          <li
            className="css-1u6i1x"
            style={
              {
                "backgroundColor": "#B15415",
                "borderColor": "#E7823D",
              }
            }
          >
            <span
              className="css-vf5797"
            >
              1234
            </span>
            <button
              aria-label="Remove "1234" tag"
              className="css-j97ryd"
              onClick={[Function]}
              type="button"
            >
              <div
                className="css-wf08df-Icon"
              >
                <svg
                  data-testid="times"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                />
              </div>
            </button>
          </li>
          <li
            className="css-1u6i1x"
            style={
              {
                "backgroundColor": "#3F2B5B",
                "borderColor": "#655181",
              }
            }
          >
            <span
              className="css-vf5797"
            >
              5678
            </span>
            <button
              aria-label="Remove "5678" tag"
              className="css-j97ryd"
              onClick={[Function]}
              type="button"
            >
              <div
                className="css-wf08df-Icon"
              >
                <svg
                  data-testid="times"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                />
              </div>
            </button>
          </li>
        </ul>
      </div>
    </div>
  </div>,
]
`;

exports[`FilterEditor component should render metric editor 1`] = `
[
  <div
    className="css-1f5gueu"
  >
    <div
      className="css-1rfunm5"
    >
      <div
        className="css-jjhwg3-input-wrapper css-rmjxhr-SelectContainer"
        onKeyDown={[Function]}
      >
        <span
          className="css-1f43avz-a11yText-A11yText"
          id="react-select-4-live-region"
        />
        <span
          aria-atomic="false"
          aria-live="polite"
          aria-relevant="additions text"
          className="css-1f43avz-a11yText-A11yText"
        />
        <div
          className="css-1scgfe8"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
        >
          <div
            className="css-1kl463j-grafana-select-value-container"
          >
            <div
              className="css-18bh10p-singleValue css-upz218-SingleValue"
            >
              edgeHitsSum
            </div>
            <div
              className=" css-lq7btu-Input"
              data-value=""
            >
              <input
                aria-autocomplete="list"
                aria-expanded={false}
                aria-haspopup={true}
                autoCapitalize="none"
                autoComplete="off"
                autoCorrect="off"
                className=""
                disabled={false}
                id="react-select-4-input"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                role="combobox"
                spellCheck="false"
                style={
                  {
                    "background": 0,
                    "border": 0,
                    "color": "inherit",
                    "font": "inherit",
                    "gridArea": "1 / 2",
                    "label": "input",
                    "margin": 0,
                    "minWidth": "2px",
                    "opacity": 1,
                    "outline": 0,
                    "padding": 0,
                    "width": "100%",
                  }
                }
                tabIndex={0}
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            className="css-uvldi-input-suffix"
          >
            <div
              className="css-wf08df-Icon"
            >
              <svg
                data-testid="angle-down"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="css-1f5gueu"
  >
    <div
      className="css-1rfunm5"
    >
      <div
        className="css-jjhwg3-input-wrapper css-rmjxhr-SelectContainer"
        onKeyDown={[Function]}
      >
        <span
          className="css-1f43avz-a11yText-A11yText"
          id="react-select-5-live-region"
        />
        <span
          aria-atomic="false"
          aria-live="polite"
          aria-relevant="additions text"
          className="css-1f43avz-a11yText-A11yText"
        />
        <div
          className="css-1scgfe8"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
        >
          <div
            className="css-1kl463j-grafana-select-value-container"
          >
            <div
              className="css-18bh10p-singleValue css-upz218-SingleValue"
            >
              Equal
            </div>
            <div
              className=" css-lq7btu-Input"
              data-value=""
            >
              <input
                aria-autocomplete="list"
                aria-expanded={false}
                aria-haspopup={true}
                autoCapitalize="none"
                autoComplete="off"
                autoCorrect="off"
                className=""
                disabled={false}
                id="react-select-5-input"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                role="combobox"
                spellCheck="false"
                style={
                  {
                    "background": 0,
                    "border": 0,
                    "color": "inherit",
                    "font": "inherit",
                    "gridArea": "1 / 2",
                    "label": "input",
                    "margin": 0,
                    "minWidth": "2px",
                    "opacity": 1,
                    "outline": 0,
                    "padding": 0,
                    "width": "100%",
                  }
                }
                tabIndex={0}
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            className="css-uvldi-input-suffix"
          >
            <div
              className="css-wf08df-Icon"
            >
              <svg
                data-testid="angle-down"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="css-1f5gueu"
  >
    <div
      className="css-1rfunm5"
    >
      <div
        className="css-p7rfez-input-wrapper"
        data-testid="input-wrapper"
      >
        <div
          className="css-1w5c5dq-input-inputWrapper"
        >
          <input
            className="css-1n6y0bv-input-input"
            onChange={[Function]}
            placeholder="Expression"
            style={
              {
                "paddingLeft": undefined,
                "paddingRight": undefined,
              }
            }
            value="1000"
          />
        </div>
      </div>
    </div>
  </div>,
]
`;
